proposals/p1927.md

```diff
@@ -0,0 +1,96 @@
+# Slice generator
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/1927)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Details](#details)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+
+<!-- tocstop -->
+
+## Problem
+
+Slices of arrays is a popular way to extract subarray of array. They also may
+be used for generating arrays.
+
+## Background
+
+There is no requirements for background.
+
+## Proposal
+
+Introduce slice generators.
+
+## Details
+
+Slice syntax is `a:b:s`. Here, `a` is the beginning of slice, `b` is the end of
+slice and `s` is the step in slice. In array indexing, `:s` can be omitted, then
+it means that step is 1. If `a` is omitted, slice starts from the beginning of
+array. The same situation is for `b`, but slice ends at the end of arrays. `:`
+is the same as `::`. If `s` is negative, lower and upper bounds are replaces
+each other.
+
+Slice generator may return `i32`, `i64`, `i128` and unsigned versions of them.
+
+Slice as generator of array:
+```carbon
+  var array: auto = (1:5:2); // array = (1,3,5)
+  var arrav: auto = 1:5:2; // arrav = (1,3,5)
+```
+Round brackets may be omitted.
```

Comment:
What is the precedence of the `:` / `::` operator? For example:

```
// Is this `(1:N) - 1`, or `1:(N - 1)`, or an error?
var x: auto = 1:N - 1;
```

Reply:
See line 54 of current version. Anyway, it should be the lowest of them for avoiding brackets while ranging in array like `array[1:N-1]`.